mindstream.el

```diff
@@ -285,40 +285,44 @@ will be opened upon loading the session.  Otherwise, follow the default
 protocol for selecting a file, including, if necessary, prompting for
 the file to be opened."
   (interactive (list ;; TODO: use completing-read-session
-                (read-directory-name "Load session: "
-                                     mindstream-save-session-path
-                                     nil
-                                     t
-                                     "")))
+                (mindstream--completing-read-session)))
   (let ((file (or file
                   (mindstream--starting-file-for-session dir))))
-    (find-file
-     (expand-file-name file
-                       dir))
+    (find-file file)
+    ;; `expand-file-name' appears to be redundant as
+    ;; `mindstream--directory-files-recursively' uses
+    ;; `directory-files-recursively', which returns file name in its
+    ;; absolute form.
+    ;; (expand-file-name file dir))
     (mindstream-begin-session)))
 
 (defun mindstream--completing-read-session ()
   "Return session-file via completion for template."
-  ;; would like this to return a folder
-  (let* ((dirs
-          (thread-last
-            (directory-files-recursively
-             mindstream-save-session-path
```

Comment:
Now `mindstream-begin-session` adds relative file name (= directory) correctly to `mindstream-session-history`, we can simplify this part.